app/models/action.rb

```diff
@@ -3,5 +3,13 @@ class Action < ActiveRecord::Base
   belongs_to :member
 
   has_paper_trail on: [:update, :destroy]
+
+  after_create :update_member_donor_status
```

Comment:
I gotta push back a bit here dude. Why is it an anti-pattern? If I go around looking for every place someone makes a donation in the controllers, I feel like it makes it much easier to leave a gap. It also means that any time someone adds a new donation workflow, they would need to know that we're tracking this counter, and need to add the appropriate code. This way, they can't mess it up.